staking/cli/src/instructions.rs

```diff
@@ -0,0 +1,472 @@
+
+    //decode tmp from base64
+    let message = base64::prelude::BASE64_STANDARD
+        .decode(encoded_message)
+        .unwrap();
+
+    let (vaa, merkle_proofs) = deserialize_accumulator_update_data(message);
+
+
+    let encoded_vaa = process_write_encoded_vaa(rpc_client, vaa.as_slice(), wormhole, payer);
+
+
+    let publisher_caps = Keypair::new();
+
+
+    init_publisher_caps(rpc_client, payer, &publisher_caps);
+
+
+    let publisher_caps_message_bytes =
+        Vec::<u8>::from(merkle_proofs.first().unwrap().message.clone());
+
+
+    for i in (0..publisher_caps_message_bytes.len()).step_by(1000) {
+        let chunk =
+            &publisher_caps_message_bytes[i..min(i + 1000, publisher_caps_message_bytes.len())];
+
+        write_publisher_caps(rpc_client, payer, publisher_caps.pubkey(), i, chunk);
```

Comment:
feels like init_puiblisher_caps + write_publisher_caps and verify_publisher_caps should all be in 1 function

Reply:
they are all used in a function `fetch_publisher_caps_and_advance`